commands.txt

```diff
@@ -1,3 +1,3 @@
 cd $SD_ARTIFACTS_DIR
 find . -type f -print > manifest.txt
-find . -type f -exec sh -c 'f=`echo $1 | sed "s/^\.\///"`; curl -H "Authorization: Bearer $SD_TOKEN" -H "Content-Type: text/plain" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$f -T ./$f' -- {} \;
+find . -type f -exec sh -c 'f=`echo $1 | sed "s/^\.\///"`; encoded=`sd-step exec core/curl "curl -s -w \"%{url_effective}\n\" --data-urlencode $f -G \"\" | sed -e \"s/^\/?//\""`; curl -H "Authorization: Bearer $SD_TOKEN" -H "Content-Type: text/plain" -X PUT $STORE_URL/v1/builds/$SD_BUILD_ID/$ARTIFACTS_DIR_SUFFIX/$encoded -T ./$f' -- {} \;
```

Comment:
Using `sd-step` every time may take long time so this should be like scm-github https://github.com/screwdriver-cd/scm-github/blob/bac74f62cb1bc937b4bb3371ef74f866b20d0313/index.js#L274

Reply:
I fixed.
Only when `--data-urlencode` option cannot be used, exec sd-step command.